spec/views/sign_up/completions/show.html.erb_spec.rb

```diff
@@ -43,12 +43,57 @@
       )
       create_identities(@user)
     end
+
     it 'informs user they are logging into an SP for the first time' do
       render
       expect(rendered).to have_content(t('titles.sign_up.new_sp'))
     end
   end
 
+  context 'signing in through an SP' do
+    let(:service_provider) do
+      create(:service_provider,
+             friendly_name: 'My Agency App',
+             agency: 'Department of Agencies')
+    end
+
+    let(:view_context) { ActionController::Base.new.view_context }
+    let(:decorated_session) do
+      ServiceProviderSessionDecorator.new(
+        sp: service_provider,
+        view_context: view_context,
+        sp_session: {
+          requested_attributes: [:email],
+        },
+        service_provider_request: ServiceProviderRequestProxy.new,
+      )
+    end
+
+    before do
+      @user.save!
```

Comment:
Could you help me understand what this is doing, and why it's necessary?

Reply:
So these tests make use of relations in the database (specifically finding identities linked to a user). The variable is set up earlier in the file with `@user = User.new` which create a User record in-memory but does not have it to the DB. Since this test needs a user in the DB, I saved it.